kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -878,26 +878,28 @@ impl<'tcx> GotocCtx<'tcx> {
     /// https://doc.rust-lang.org/std/intrinsics/fn.ptr_offset_from.html
     fn codegen_ptr_offset_from(
         &mut self,
+        intrinsic: &str,
         mut fargs: Vec<Expr>,
         p: &Place<'tcx>,
         loc: Location,
     ) -> Stmt {
-        let a = fargs.remove(0);
-        let b = fargs.remove(0);
-        let pointers_to_same_object = a.clone().same_object(b.clone());
+        let dst_ptr = fargs.remove(0);
+        let src_ptr = fargs.remove(0);
 
-        Stmt::block(
-            vec![
-                self.codegen_assert(
-                    pointers_to_same_object,
-                    PropertyClass::PointerOffset,
-                    "ptr_offset_from: pointers point to same object",
-                    loc.clone(),
-                ),
-                self.codegen_expr_to_place(p, a.sub(b)),
-            ],
-            loc,
-        )
+        // Check that the computation would not overflow an `isize`
```

Comment:
Why did you remove the `same_object` check

Reply:
The `same_object` check is being done by CBMC by default. I converted the test we already had for it into an `expected` test so we ensure the "same object violation" stays there.